src/protocols/secure_channel/CASESession.cpp

```diff
@@ -172,23 +185,31 @@ CHIP_ERROR CASESession::Init(SessionManager & sessionManager, Credentials::Certi
 }
 
 CHIP_ERROR
-CASESession::ListenForSessionEstablishment(SessionManager & sessionManager, FabricTable * fabrics,
+CASESession::ListenForSessionEstablishment(SessionManager & sessionManager, FabricTable * fabricTable,
                                            SessionResumptionStorage * sessionResumptionStorage,
                                            Credentials::CertificateValidityPolicy * policy, SessionEstablishmentDelegate * delegate,
                                            Optional<ReliableMessageProtocolConfig> mrpConfig)
 {
-    VerifyOrReturnError(fabrics != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(fabricTable != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
     ReturnErrorOnFailure(Init(sessionManager, policy, delegate));
 
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    SuccessOrExit(err = fabricTable->AddFabricDelegate(&mFabricDelegate));
```

Comment:
Should this not be done only after we get the Sigma1? Maybe it makes no difference, but what happens if `AbortPendingEstablish` is called while not yet establishing? We should probably have a unit test for that

Reply:
Maybe it would be a little more clear if I did this in `Init()`, but I was under the impression that we want to do this whenever we haver essentially already done `AllocateSecureSession()` (that is in `Init`)